src/big/big_float.cr

```diff
@@ -17,6 +18,23 @@ struct BigFloat < Float
     LibGMP.mpf_init_set_str(out @mpf, str, 10)
   end
 
+  def initialize(num : BigInt)
```

Comment:
@akzhan, sorry, either you do it wrong way, or Crystal does it wrong way.
In fact, your failed tests were just wrong, cause you assumed `BigFloat.new(some_string).to_s == some_string`, but it is wrong! BigFloat - is not decimal float, but binary float.